cmake/onnxruntime.cmake

```diff
@@ -195,3 +210,47 @@ set_target_properties(onnxruntime PROPERTIES FOLDER "ONNXRuntime")
 if (WINDOWS_STORE)
   target_link_options(onnxruntime PRIVATE /DELAYLOAD:api-ms-win-core-libraryloader-l1-2-1.dll)
 endif()
+
+# Assemble the Apple static framework (iOS and macOS)
+if(onnxruntime_BUILD_APPLE_FRAMEWORK)
+  set(STATIC_LIB_DIR ${CMAKE_CURRENT_BINARY_DIR}/static_libraries)
+  file(MAKE_DIRECTORY ${STATIC_LIB_DIR})
+
+  # Remove the existing files in the STATIC_LIB_DIR folder
+  file(GLOB _OLD_STATIC_LIBS ${STATIC_LIB_DIR}/*.a)
+  file(REMOVE "${_OLD_STATIC_LIBS}")
+
+  # Go through all the static libraries, and create symbolic links
+  foreach(_LIB ${onnxruntime_INTERNAL_LIBRARIES} ${onnxruntime_EXTERNAL_LIBRARIES})
+    GET_TARGET_PROPERTY(_LIB_TYPE ${_LIB} TYPE)
+    if(_LIB_TYPE STREQUAL "STATIC_LIBRARY")
+      add_custom_command(TARGET onnxruntime POST_BUILD COMMAND ${CMAKE_COMMAND} -E create_symlink $<TARGET_FILE:${_LIB}> ${STATIC_LIB_DIR}/$<TARGET_LINKER_FILE_NAME:${_LIB}>)
+    endif()
+  endforeach()
+
+  if(${CMAKE_SYSTEM_NAME} STREQUAL "iOS")
+    set(STATIC_FRAMEWORK_OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_BUILD_TYPE}-${CMAKE_OSX_SYSROOT})
+  else() # macOS
+    set(STATIC_FRAMEWORK_OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR})
+  endif()
+
+  # Assemble the static framwwork
+  set(STATIC_FRAMEWORK_DIR ${STATIC_FRAMEWORK_OUTPUT_DIR}/static_framework/onnxruntime.framework)
+  set(STATIC_FRAMEWORK_HEADER_DIR ${STATIC_FRAMEWORK_DIR}/Headers)
+  file(MAKE_DIRECTORY ${STATIC_FRAMEWORK_DIR})
+  # Remove all files under STATIC_FRAMEWORK_DIR (if any)
+  file(GLOB_RECURSE _OLD_STATIC_FRAMEWORK ${STATIC_FRAMEWORK_DIR}/*.*)
+  file(REMOVE "${_OLD_STATIC_FRAMEWORK}")
+
+  file(MAKE_DIRECTORY ${STATIC_FRAMEWORK_HEADER_DIR})
+
+  # copy the header files one by one, and the Info.plist
+  foreach(h_ ${APPLE_FRAMEWORK_HEADERS})
+    get_filename_component(HEADER_NAME_ ${h_} NAME)
+    configure_file(${h_} ${STATIC_FRAMEWORK_HEADER_DIR}/${HEADER_NAME_} COPYONLY)
```

Comment:
>configure_file [](http://example.com/codeflow?start=4&length=14)

will these get updated if the header files change?
seems like it only runs when build files are generated.

Reply:
Move to post build command, also changed the Android header copy